src/java.net.http/share/classes/jdk/internal/net/http/common/Utils.java

```diff
@@ -1123,13 +1113,16 @@ public static String encode(String s) {
             assert false : x;
         }
 
+        HexFormat format = HexFormat.of().withUpperCase();
```

Comment:
Its a bit of a space/time trade-off.  And the balance point will change if the class is converted in the future to a Valhalla primitive class. At this point it looks to me like a premature optimization.

Reply:
OK - but just to avoid misunderstanding - I meant a static final field in  jdk.internal.net.http.common.Utils; as in
```
public class Utils { ...
    private static final HexFormat HEX_FORMAT = HexFormat.of().withUpperCase();
    ...
}
```

Reply:
Yes, that's what I understood.  Still it take up space and gc is aware of it.